tests/longhaul/base_block_test.go

```diff
@@ -0,0 +1,278 @@
+package longhaul
+
+import (
+	"sync"
+	"testing"
+
+	"github.com/coreos/pkg/capnslog"
+	"github.com/rook/rook/tests/framework/installer"
+	"github.com/rook/rook/tests/framework/utils"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+var (
+	logger           = capnslog.NewPackageLogger("github.com/rook/rook", "longhaul")
+	defaultNamespace = "default"
+)
+
+func setUpRookAndPoolInNamespace(t func() *testing.T, namespace string, storageClassName string, poolName string) (*utils.K8sHelper, *installer.InstallHelper) {
+	kh, err := utils.CreateK8sHelper(t)
+	assert.Nil(t(), err)
+
+	installer := installer.NewK8sRookhelper(kh.Clientset, t)
+	if !kh.IsRookInstalled(namespace) {
+		isRookInstalled, err := installer.InstallRookOnK8sWithHostPath(namespace, "bluestore", "/temp/rookBackup")
+		require.NoError(t(), err)
+		require.True(t(), isRookInstalled)
+	}
+
+	//create storage class
+	if scp, _ := kh.IsStorageClassPresent(storageClassName); !scp {
+		logger.Infof("Install storage class for rook block")
+		_, err := storageClassOperation(kh, storageClassName, poolName, namespace, "create")
+		require.NoError(t(), err)
+
+		//make sure storageclass is created
+		present, err := kh.IsStorageClassPresent(storageClassName)
+		require.NoError(t(), err)
+		require.True(t(), present, "Make sure storageclass is present")
+	}
+	return kh, installer
+}
+
+// Set Up rook, storageClass ,pvc and mysql pods for longhaul test
+// All the set up is if needed.
+func createPVCAndMountMysqlPod(t func() *testing.T, kh *utils.K8sHelper, storageClassName string, appName string, appLabel string, pvcName string) *utils.MySQLHelper {
+
+	//create mysql pod
+	if _, err := kh.GetPVCStatus(defaultNamespace, pvcName); err != nil {
+		logger.Infof("Create PVC")
+
+		mySqlPodOperation(kh, storageClassName, appName, appLabel, pvcName, "create")
+
+		//wait till mysql pod is up
+		require.True(t(), kh.IsPodInExpectedState(appLabel, "", "Running"))
+		require.True(t(), kh.WaitUntilPVCIsBound(defaultNamespace, pvcName))
+	}
+	dbIP, err := kh.GetPodHostID(appLabel, "")
+	require.Nil(t(), err)
+	dbPort, err := kh.GetServiceNodePort(appName, "default")
+	require.Nil(t(), err)
+	//create database connection
+	db := utils.CreateNewMySQLHelper("mysql", "mysql", dbIP+":"+dbPort, "sample")
+
+	require.True(t(), db.PingSuccess())
+
+	if exist := db.TableExists(); !exist {
+		db.CreateTable()
+	}
+
+	return db
+
+}
+
+func storageClassOperation(k8sh *utils.K8sHelper, storageClassName string, poolName string, namespace string, action string) (string, error) {
```

Comment:
should these methods go in a shared package so integration tests can use them too?

Reply:
Integration tests have these helper too, but the load test are a little different and i wanted to keep them saperated